x-pack/solutions/security/plugins/security_solution/public/siem_migrations/rules/service/rule_migrations_service.ts

```diff
@@ -119,21 +120,41 @@ export class SiemRulesMigrationsService {
       });
   }
 
-  public async createRuleMigration(body: CreateRuleMigrationRequestBody): Promise<string> {
-    const rulesCount = body.length;
+  public async addRulesToMigration(
+    migrationId: string,
+    rules: CreateRuleMigrationRulesRequestBody
+  ) {
+    const rulesCount = rules.length;
     if (rulesCount === 0) {
       throw new Error(i18n.EMPTY_RULES_ERROR);
     }
 
     try {
-      let migrationId: string | undefined;
       // Batching creation to avoid hitting the max payload size limit of the API
       for (let i = 0; i < rulesCount; i += CREATE_MIGRATION_BODY_BATCH_SIZE) {
-        const bodyBatch = body.slice(i, i + CREATE_MIGRATION_BODY_BATCH_SIZE);
-        const response = await createRuleMigration({ migrationId, body: bodyBatch });
-        migrationId = response.migration_id;
+        const rulesBatch = rules.slice(i, i + CREATE_MIGRATION_BODY_BATCH_SIZE);
+        await addRulesToMigration({ migrationId, body: rulesBatch });
       }
       this.telemetry.reportSetupMigrationCreated({ migrationId, rulesCount });
+    } catch (error) {
+      this.telemetry.reportSetupMigrationCreated({ migrationId, rulesCount, error });
+      throw error;
+    }
+  }
+
+  public async createRuleMigration(data: CreateRuleMigrationRulesRequestBody): Promise<string> {
+    const rulesCount = data.length;
+    if (rulesCount === 0) {
+      throw new Error(i18n.EMPTY_RULES_ERROR);
+    }
+
+    try {
+      // create the migration
+      const { migration_id: migrationId } = await createRuleMigration({});
+
+      await this.addRulesToMigration(migrationId, data);
+
+      this.telemetry.reportSetupMigrationCreated({ migrationId, rulesCount });
```

Comment:
We are sending this telemetry twice, here and inside the `addRulesToMigration`. 
Maybe we should send two different events, but since (for now) the UI does not allow calling _addRules_ to a created migration. We can just remove the one inside the `addRulesToMigration`.

Reply:
Thanks.Corrected.